source/AI.cpp

```diff
@@ -3824,8 +3901,8 @@ void AI::IssueOrders(const PlayerInfo &player, const Orders &newOrders, const st
 
 	// A target is valid if we have no target, or when the target is in the
 	// same system as the flagship.
-	bool isValidTarget = !newTarget || (newTarget && player.Flagship() &&
-		newTarget->GetSystem() == player.Flagship()->GetSystem());
+	bool isValidTarget = !targetShip || targetAsteroid
```

Comment:
Also here: Check if the asteroid is in the same system.

Reply:
Per discussion in discord #the-hub between quyykk and I this is no longer necessary.